From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Thu, 1 Jun 2023 19:30:16 +0200
Subject: [PATCH] Configuration - Get Fiddle configuration with Bukkit API

Allows getting the Fiddle configuration using the Bukkit API.
This is done by implementing `Server.getFiddleConfig()`, which is defined in the "Get Fiddle configuration with Bukkit API" patch in the `fiddle-api` module, in `CraftServer`.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

This patch is based on the following patch:
"Gale configuration"
By: Martijn Muijsers <martijnmuijsers@live.nl>
As part of: Gale (https://github.com/GaleMC/Gale)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

* Gale description *

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Gale - https://galemc.org

This patch is based on the following patch:
"Paper config files"
By: Jake Potrebic <jake.m.potrebic@gmail.com>
As part of: Paper (https://github.com/PaperMC/Paper)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 48592fe72678f2e4121b5b11cc6d30ae146c19fa..cb68dc486da8519b410859ace508dcf9e3adda5e 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2865,6 +2865,14 @@ public final class CraftServer implements Server {
             return CraftServer.this.console.paperConfigurations.createLegacyObject(CraftServer.this.console);
         }
 
+        // Fiddle start - Fiddle configuration - API
+        @Override
+        public YamlConfiguration getFiddleConfig()
+        {
+            return FiddleConfigurations.get(null).createLegacyObject(CraftServer.this.console);
+        }
+        // Fiddle end - Fiddle configuration - API
+
         @Override
         public void restart() {
             org.spigotmc.RestartCommand.restart();
